### PR TITLE
Add channel picker to gift card create

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it adds channel picker to gift card create.

⚠️ Contains changes from https://github.com/mirumee/saleor-dashboard/pull/1332.

**PR intended to be tested with API branch:** 

### Screenshots

Zrzut ekranu 2021-08-30 o 12 44 02

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://feature-gift-cards-pre-mvp.api.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4264 Add channel picker to gift card create modal](https://app.clickup.com/t/2549495/SALEOR-4264)

File: src/giftCards/components/GiftCardSendToCustomer/messages.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  customerSubtitle: {
+    defaultMessage:
+      "Selected customer will be sent the generated gift card code. Someone else can redeem the gift card code. Gift card will be assigned to account which redeemed the code.",
+    description: "GiftCardCreateDialog customer subtitle"
+  },
+  customerChannelSubtitle: {
+    defaultMessage:
+      "Customer will be sent the gift card code via this channels email address",
+    description: "GiftCardCreateDialog customer channel subtitle"
```
Comment (dominik-zeglen) [inline src/giftCards/components/GiftCardSendToCustomer/messages.ts:16]: `GiftCardCreateDialog` doesn't tell anything useful to translator

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

File: src/giftCards/components/GiftCardExpirySelect/GiftCardExpirySelect.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  expirySelected: boolean;
+  expiryPeriodType: TimePeriodTypeEnum;
+  expiryPeriodAmount: number;
+  expiryType: GiftCardExpiryType;
+  customOptions?: UntranslatedOption[];
+  errors?: Record<"expiryDate", GiftCardError>;
+  expiryDate?: string;
+}
+
+const GiftCardExpirySelect: React.FC<GiftCardExpirySelectProps> = ({
```
Comment (mmarkusik) [inline src/giftCards/components/GiftCardExpirySelect/GiftCardExpirySelect.tsx:47]: About this component - we have 3 places now where we do this sort of selection:
- gift card settings
Screenshot 2021-09-06 at 13 35 26

- gift card update 
Screenshot 2021-09-06 at 13 35 37

- gift card create
Screenshot 2021-09-06 at 13 35 45

As we can see on the screens, they're all pretty much different, regarding strings, substrings, custom content (like the displayed date in create dialog) options to show in each one etc. That's why I'd rather keep them separate components instead of generic components with custom options etc.

TLDR; please change this component (logic, name and placement in the dir) to GiftCardCreateExpirySelect, remove the custom options etc. If needed, also make one for GiftCardSettingsExpirySelect

**Review by mmarkusik — CHANGES_REQUESTED** [review]
